### ISSUE TITLE
[Tables] updateOne with slice condition on the clustering columns returns HTTP 500 and an internal server error

### ISSUE THREAD

**Issue by hemidactylus** [stargate/data-api#1758]
Table:

```
CREATE TABLE default_keyspace.x (
    a int,
    b int,
    x int,
    PRIMARY KEY (a, b)
) WITH CLUSTERING ORDER BY (b ASC)
```

updateOne payload:

```
{
    "updateOne": {
        "filter": {
            "a": 10,
            "b": {"$gt": 1}
        },
        "update": {"$set": {"x": 1}}
    }
}
```

The API returns with a 500 instead of a controlled 200-with-error (as it should):

```
{
    "errors": [
        {
            "message": "An unexpected server error occurred while processing the request. \n\nError Class: InvalidQueryException\nError Message: Slice restrictions are not supported on the clustering columns in UPDATE statements",
            "scope": "",
            "errorCode": "UNEXPECTED_SERVER_ERROR",
            "id": "3dffdbc6-92a3-4a44-a3ce-105e4577c557",
            "family": "SERVER",
            "title": "Unexpected server error"
        }
    ]
}
```

**Comment by amorton** [comment]
we should retest this after the fix for #1729 I think we made it return a better error there

**Comment by Hazel-Datastax** [comment]
Confirmed that the issue has been fixed. Current error message is:

```json
{
    "errors": [
        {
            "message": "The Filtering using the non `$eq` filter operations can select more than one row, and so cannot be used with the `updateOne` and `deleteOne` commands as they can only modify one row.\n\nThe command used an invalid filter on the columns: b.\n\nResend the command only using `$eq` filters.",
            "scope": "FILTER",
            "errorCode": "UNSUPPORTED_FILTER_FOR_UPDATE_ONE_DELETE_ONE",
            "exceptionClass": "FilterException",
            "id": "e62e51f1-6b1a-4438-b727-75a79fa6fffc",
            "family": "REQUEST",
            "title": "updateOne and deleteOne commands only support filtering using $eq"
        }
    ]
}
```
